catalog/plugin/catalog-plugin-querymonitor/src/main/java/ddf/catalog/querymonitor/ActiveSearch.java

```diff
@@ -0,0 +1,200 @@
+                    + "s %4$-" + TIME_PAD_LENGTH + "." + TIME_MAX_LENGTH + "s %5$-"
+                    + UUID_PAD_LENGTH + "." + UUID_MAX_LENGTH + "s %n";
+
+    public String getCQL() {
+        return cqlQuery;
+    }
+
+    /**
+     * Returns a {@link Date} representing the approximate time this search started.
+     *
+     * @return {@link Date} representing the approximate time the search started.
+     */
+    public Date getStartTime() {
+        return (Date) startTime.clone();
+    }
+
+    /**
+     * Returns the source information for this {@link ActiveSearch}
+     *
+     * @return {@link Source} representing the source being queried by this {@link ActiveSearch}.
+     * Can return {@code null} when the object was created using a null source parameter to the constructor.
+     */
+    public Source getSource() {
+        return source;
+    }
+
+    /**
+     * Returns the {@link UUID} for this {@link ActiveSearch}
+     *
+     * @return {@link UUID} representing the unique identifier for this {@link ActiveSearch}.
+     */
+    public UUID getUniqueID() {
+        return uniqueID;
+    }
+
+    /**
+     * Returns the client information for this {@link ActiveSearch}
+     *
+     * @return {@link String} representing the client info that queried the {@link Source}.
+     */
+    public String getClientInfo() {
+        return clientInfo;
+    }
+
+    /**
+     * Constructor for an {@link ActiveSearch}.
+     *
+     * @param cqlQuery {@link String} that represents the query.
+     * @param source   {@link Source} that the search is querying.
+     * @param uniqueID {@link UUID} a unique ID that is used to uniquely identify the search.
+     */
+
+    public ActiveSearch(String cqlQuery, Source source, UUID uniqueID, String clientInfo) {
+        if (cqlQuery == null) {
+            cqlQuery = "unknownCQLQuery";
+            LOGGER.debug("CQLQuery for new ActiveSearch is null. Being set as 'unknownCQLQuery'.");
+        }
+        this.cqlQuery = cqlQuery;
+        if (source == null) {
+            LOGGER.debug("Source for new ActiveSearch is null.");
+        }
+        this.source = source;
+        startTime = new Date();
+        if (uniqueID == null) {
+            this.uniqueID = java.util.UUID.randomUUID();
+            LOGGER.warn("uniqueID for new ActiveSearch is null. Setting the ID as {}", uniqueID);
+        } else {
+            this.uniqueID = uniqueID;
+        }
+        if (clientInfo == null) {
+            this.clientInfo = "unknownClient";
+        } else {
+            this.clientInfo = clientInfo;
+        }
+    }
+
+    private static String getCqlFromQueryRequest(QueryRequest request) {
+        String cqlQuery;
+        if (request == null) {
+            return "";
+        } else {
+            cqlQuery = request.toString();
+            String tempString = StringUtils.substringBetween(cqlQuery, "filter=[", " ],startIndex");
```

Comment:
I am wondering if this would work better: http://docs.geotools.org/latest/javadocs/org/geotools/filter/text/cql2/CQL.html#toCQL-org.opengis.filter.Filter-